core/main/src/state/bootstrap_state.rs

```diff
@@ -107,9 +109,42 @@ impl BootstrapState {
                 .expect("Valid app manifest");
         let extn_manifest = LoadExtnManifestStep::get_manifest();
         let extn_state = ExtnState::new(channels_state.clone(), extn_manifest.clone());
-        let platform_state =
-            PlatformState::new(extn_manifest, device_manifest, client, app_manifest_result);
-
+        let platform_state = PlatformState::new(
+            extn_manifest,
+            device_manifest,
+            client,
+            app_manifest_result,
+            ripple_version_from_etc(),
+        );
+
+        fn ripple_version_from_etc() -> Option<String> {
```

Comment:
You may change this inner function as a closure 
`let ripple_version_from_etc = || -> Option<String>  {`